cityofaustin/atd-moped
# Normalize project status (part 3)

## Associated issues
https://github.com/cityofaustin/atd-data-tech/issues/7910

See also #871, #872.

This PR focuses on tightening up the editor interface with the `moped_proj_phases` table, and addresses some DB schema cleanup made possible by the earlier stages of this work. Nearly of the visual/front-end changes from this PR affect the "Phases" table on the "Timeline" tab and the dashboard status modal.


It does these things to the project phases table:
- deprecates the use of an "Unknown" (id `0`) phase and subphase lookup value in favor of good ole `null`
- Uses an autocomplete input for the Phase column input, which is now rather long with the addition of "On hold" and "Canceled"
- Adds stateful tracking of if a project phase is being mutated, and uses material-table's `isLoading` prop to display a loading spinner between edit actions until all mutations and refetching are complete.
- Adds MUI chip styling to the current phase
- Renames a few column headers

It also:
- Uses a migration to insert `Canceled` and `On hold` `moped_proj_phases` on projects where those statuses were being used.
- Drops the `moped_status` table
- Establishes a proper relationship between `moped_proj_phases.subphase_id` and the `moped_subphases` table.
- Reboots the seed data and adds documentation about how to maintain it 😬 




## Testing
**URL to test:**  Local

**Steps to test:**
Definitely test with new seed data. Would also be good to test with a [prod replica](https://austininnovation.slack.com/archives/CNUEPKLB1/p1668447762094979).

1. In the project list view, notice all projects have a status badge.
2. Edit projects phases - trying very hard to break the feature. 
- Add multiple phases with all fields filled out
- Edit a phase by changing it's phase name and subphase. Edit other fields
- Toggle current phase back and forth on different phases
- Delete phases
- While interacting with phase editing, notice that table displays a loading spinner while rows are being mutated
3. Create a new project and make sure it has a Potential current phase 
4. Checkout the dashboard tab and make sure status badges look good. Add / edit phases from the status modal.
5. Add a subproject to a project and make sure it's badge renders in the subprojects table.
6. Before you stop the cluster, try following the steps in README_seed_data. Spin up the cluster to view your persisted changes.

---
#### Ship list
